script.js

```diff
@@ -1,3 +1,83 @@
+// Vanilla JS debounce function, by Josh W. Comeau:
+// https://www.joshwcomeau.com/snippets/javascript/debounce/
+function debounce(callback, wait) {
+  let timeoutId = null;
```

Comment:
There's no usage of es6 syntax anywhere in the file so far so far.
Worth noticing that we should be ok with browser support as we no longer support IE11, but in case there are still users relying on that browser (or older versions of safari) this will throw invalid syntax errors.

Reply:
@luis-almeida I see what you're saying. I leaned into ES6 (here, and in a few other places) because we're very clear that we no longer support IE11. By saying
> in case there are still users relying on that browser (or older versions of safari) this will throw invalid syntax errors.
that seems to imply that we _do_ actually have a need to support older browsers, though. Is there a team stance on this?

Reply:
I agree with you. Sorry if my comment was confusing.
What I mean is that we no longer have to support older browsers so we should definitely be ok with this change. But since we have only been using es5 syntax so far, it means that there were no syntax errors only runtime errors.
This change will lead to syntax errors in IE11 which might remove support for all JS based features in the theme. We should be aware of that in case some customers complain.